src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1986,64 +1965,52 @@ else if (compareAndSetCtl(c, c) && casRunState(e, e | STOP))
     final void runWorker(WorkQueue w) {
         if (w != null) {
             int phase = w.phase, r = w.stackPred;     // seed from registerWorker
-            int cfg = w.config, src = -1, nsteals = 0;
-            rescan: for (boolean scanned = false;;) {
+            int cfg = w.config, nsteals = 0, src = -1;
+            for (;;) {
```

Comment:
Could the non-labeled version introduce different GC safepoints which could have adverse impacts? 🤔

Reply:
Yes, one of many uncertainties and tradeoffs...